src/Microsoft.CodeQuality.Analyzers.Exp/Core/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAbstractValue.cs

```diff
@@ -17,11 +17,9 @@ internal class PointsToAbstractValue: IEquatable<PointsToAbstractValue>
         public static PointsToAbstractValue NoLocation = new PointsToAbstractValue(PointsToAbstractValueKind.NoLocation);
         public static PointsToAbstractValue Unknown = new PointsToAbstractValue(PointsToAbstractValueKind.Unknown);
         
-        public PointsToAbstractValue(ImmutableHashSet<AbstractLocation> locations, PointsToAbstractValueKind kind)
+        private PointsToAbstractValue(ImmutableHashSet<AbstractLocation> locations, PointsToAbstractValueKind kind)
         {
-            Debug.Assert(kind != PointsToAbstractValueKind.Known || !locations.IsEmpty);
-            Debug.Assert(kind != PointsToAbstractValueKind.NoLocation || locations.IsEmpty);
-            Debug.Assert(kind != PointsToAbstractValueKind.Undefined || locations.IsEmpty);
+            Debug.Assert(locations.IsEmpty == (kind != PointsToAbstractValueKind.Known));
```

Comment:
While I can appreciate the smaller number of asserts, it took me a minute of mental puzzling to figure out what this assert actually means. I _much_ prefer the original asserts.

Reply:
Sure, will make this more cleaner in next PR. Thanks.